lib/manager/nuget/extract.ts

```diff
@@ -1,19 +1,80 @@
+import findUp from 'find-up';
+import * as path from 'path';
+import { XmlDocument } from 'xmldoc';
+import { readFile } from 'fs-extra';
 import { logger } from '../../logger';
 import { get } from '../../versioning';
 import { PackageDependency, ExtractConfig, PackageFile } from '../common';
 import * as semverVersioning from '../../versioning/semver';
 import * as datasourceNuget from '../../datasource/nuget';
 import { SkipReason } from '../../types';
 
-export function extractPackageFile(
+async function readFileAsXmlDocument(file: string): Promise<XmlDocument> {
+  try {
+    return new XmlDocument((await readFile(file)).toString());
+  } catch (err) {
+    logger.debug({ err }, `failed to parse '${file}' as XML document`);
+    return undefined;
+  }
+}
+
+async function determineRegistryUrls(
+  packageFile: string,
+  localDir: string
+): Promise<string[]> {
+  const nuGetConfigPath = await findUp('NuGet.config', {
+    cwd: path.dirname(path.join(localDir, packageFile)),
+    type: 'file',
+  });
```

Comment:
Can this throw somehow?

Reply:
I don't think so: https://github.com/sindresorhus/find-up#findupname-options